docs/concepts/README.md

```diff
@@ -0,0 +1,217 @@
+# Duck Typing
+
+Knative uses duck typing to keep various components loosely coupled, but
+[what is duck typing](https://en.wikipedia.org/wiki/Duck_typing)? It is
+reasoning about a resource's control plane shape and behaviors based on a common
+definition of that contact. If a resource has the same fields in the same schema
+locations as the contract specifies, and the control/data plane behaviors as the
+contract specifies, Knative can use that resource as if it is the generic duck
+type with little specific knowledge about the resource type. Some resources may
+choose to opt-in or be opt'ed-in to multiple duck types.
+
+<!-- TODO: point to Discovery ClusterDuckType documentation. -->
```

Comment:
I'd make Discovery a different topic to Duck types under the Concepts folder, like I mentioned in the other comment, re not having this all in the index file. I think it'll be easier to read if it's a page per topic.